packages/jest-circus/src/event_handler.js

```diff
@@ -43,33 +43,42 @@ const handler: EventHandler = (event, state): void => {
     }
     case 'add_hook': {
       const {currentDescribeBlock} = state;
-      const {fn, hookType: type, timeout} = event;
+      const {asyncError, fn, hookType: type, timeout} = event;
```

Comment:
i was thinking that maybe we can name it somehow differently?
when i see `asyncError` i assume that it already happened, but in this case we're just capturing it to use in the future?

not in this PR, but maybe something to change in the future

Reply:
Yeah, it's just so we can keep the stack across async boundaries

Reply:
Open to naming suggestions